src/YAMLValidator/YamlLoader.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.ObjectModel;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence;
+internal sealed class YamlLoader
```

Comment:
see all my comments on `SchemaLoader`

Reply:
I'm getting this error when I make the class stateless and the function non static: `CA1822: Mark members as static`, 'Member Load does not access instance data" -> Its fine if I just supress it, right?

```
    public IReadOnlyDictionary<string, string> Load(string filePath, string pathType)
    {
        var loadedYaml = LoadFilePathData(filePath, pathType);
        return new ReadOnlyDictionary<string, string>(loadedYaml);
    }

```